C/plugins/storage/common/disk_monitor.cpp

```diff
@@ -0,0 +1,159 @@
+/*
+ * Fledge storage service.
+ *
+ * Copyright (c) 2024 Dianomic Systems
+ *
+ * Released under the Apache 2.0 Licence
+ *
+ * Author: Mark Riddoch
+ */
+#include <disk_monitor.h>
+#include <logger.h>
+#include <sys/vfs.h>
+#include <string.h>
+
+using namespace std;
+
+/**
+ * Construt a disk space monitor class
+ *
+ * It monitors the free space on two path, since the storage service may use
+ * different locations for reading and configuration storage. If they are both
+ * the same file system then the monitor is only done once.
+ *
+ * If the free space falls below 5% a fatal error is written to the error log
```

Comment:
5% and 10% may be too late for the cases where disk is getting filled too fast e.g. with vibration data? Also some reasonable disk space may need to be kept aside for system related files to avoid corrupting the system files.

Reply:
Hence the prediction stuff. Although I would say if you are ingesting that sort of data then you should have a biggish storage system anyway. In which case 10% is a lot of storage.